aws-core/src/main/java/software/amazon/awssdk/awscore/exception/AwsErrorDetails.java

```diff
@@ -0,0 +1,266 @@
+
+    /**
+     * @return the human-readable error message provided by the service.
+     */
+    public String errorMessage() {
+        return errorMessage;
+    }
+
+    /**
+     * Returns the error code associated with the response.
+     */
+    public String errorCode() {
+        return errorCode;
+    }
+
+    /**
+     * Returns the response payload as bytes.
+     */
+    public byte[] rawResponse() {
+        return rawResponse == null ? null : rawResponse.clone();
+    }
+
+    /**
+     * Returns a map of HTTP headers associated with the error response.
+     */
+    public Map<String, String> headers() {
+        return headers;
+    }
+
+    /**
+     * @return {@link SdkException.Builder} instance to construct a new {@link AwsErrorDetails}.
+     */
+    public static Builder builder() {
+        return new BuilderImpl();
+    }
+
+    /**
+     * Create a {@link AwsErrorDetails.Builder} initialized with the properties of this {@code AwsErrorDetails}.
+     *
+     * @return A new builder initialized with this config's properties.
+     */
+    public Builder toBuilder() {
+        return new BuilderImpl(this);
+    }
+
+    public static Class<? extends Builder> serializableBuilderClass() {
+        return BuilderImpl.class;
+    }
+
+    public interface Builder {
+        /**
+         * Specifies the error message returned by the service.
+         *
+         * @param errorMessage The error message returned by the service.
+         * @return This object for method chaining.
+         */
+        Builder errorMessage(String errorMessage);
+
+        /**
+         * The error message specified by the service.
+         *
+         * @return The error message specified by the service.
+         */
+        String errorMessage();
+
+        /**
+         * Specifies the error code returned by the service.
+         *
+         * @param errorCode The error code returned by the service.
+         * @return This object for method chaining.
+         */
+        Builder errorCode(String errorCode);
+
+        /**
+         * The error code specified by the service.
+         *
+         * @return The error code specified by the service.
+         */
+        String errorCode();
+
+        /**
+         * Specifies the name of the service that returned this error.
+         *
+         * @param serviceName The name of the service.
+         * @return This object for method chaining.
+         */
+        Builder serviceName(String serviceName);
+
+        /**
+         * The name of the service that returned this error.
+         *
+         * @return The name of the service that returned this error.
+         */
+        String serviceName();
+
+        /**
+         * Specifies the headers returned on the error response from the service.
+         *
+         * @param headers A map of headers from the response.
+         * @return This object for method chaining.
+         */
+        Builder headers(Map<String, String> headers);
```

Comment:
Can we just return the SdkHttpResponse so they can get the HTTP error code / message as well?

Reply:
The error response handlers only have access to HttpResponse currently and not SdkHttpResponse.